src/RamOperation.cpp

```diff
@@ -202,17 +202,17 @@ std::unique_ptr<RamValue> RamAggregate::getIndexElement(RamCondition* c, size_t&
     if (auto* binRelOp = dynamic_cast<RamConstraint*>(c)) {
         if (binRelOp->getOperator() == BinaryConstraintOp::EQ) {
             if (auto* lhs = dynamic_cast<RamElementAccess*>(binRelOp->getLHS())) {
-                RamValue* rhs = binRelOp->getRHS();
+                auto* rhs = binRelOp->getRHS()->clone();
```

Comment:
Make rhs a unique pointer to avoid the clone being accidentally not deleted, as happens here if the condition on the next line is not met.

Better yet, don't clone it until you know you need the clone.

Reply:
Thanks this was a bad mistake on my part. D: